tests/sockets_cores_threads_test.go

```diff
@@ -0,0 +1,493 @@
+			Expect(err).ToNot(HaveOccurred())
+
+			err = xml.Unmarshal([]byte(vmiXml122), domStat122)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("check XML")
+
+			By("check XML - vmi121")
+			Expect(uint(domStat121.VCPU.CPUs) == 2).To(BeTrue())
+			Expect(uint(domStat121.CPU.Topology.Sockets) == 2).To(BeTrue())
+			Expect(uint(domStat121.CPU.Topology.Cores) == 1).To(BeTrue())
+			Expect(uint(domStat121.CPU.Topology.Threads) == 1).To(BeTrue())
+
+			By("check XML - vmi122")
+			Expect(uint(domStat122.VCPU.CPUs) == 2).To(BeTrue())
+			Expect(uint(domStat122.CPU.Topology.Sockets) == 2).To(BeTrue())
+			Expect(uint(domStat122.CPU.Topology.Cores) == 1).To(BeTrue())
+			Expect(uint(domStat122.CPU.Topology.Threads) == 1).To(BeTrue())
+
+			By("Clean old pods")
+			var cleanPods []*corev1.Pod
+			clean_pods(virtClient, cleanPods)
+
+		})
+	})
+
+	Context("test cases 1.3,2.1,2.2 Check the validity of the XML file if user didn’t set the CPU topology at all", func() {
+		It("[test_id:1488] [test_id:1490] [test_id:1489] [test_id:1488] testcases 1.3,2.1,2.2 Check the validity of the XML file if user didn’t set the CPU topology at all", func() {
+
+			var wg sync.WaitGroup
+			vm_index := 0
+
+			for sockets := 0; sockets < 3; sockets++ {
+				for cores := 0; cores < 3; cores++ {
+					for threads := 0; threads < 3; threads++ {
+
+						By("1.3 check resources in the cluster")
+						cpuNeeded := 1
+						if sockets > 0 {
+							cpuNeeded *= sockets
+						}
+						if cores > 0 {
+							cpuNeeded *= cores
+						}
+						if threads > 0 {
+							cpuNeeded *= threads
+						}
+						const memNeeded int64 = 256 * 1024 * 1024 // 256mb is default in the template
+
+						isAvailable := false
+						const maxWaitIterations = 15 // half of minute
+						var cpuLimit, memLimit int
+						var IsResourcesInCluster bool
+						for i := 0; i < maxWaitIterations; i++ {
+							IsResourcesInCluster, cpuLimit, memLimit = isEnoughResources(virtClient, cpuNeeded, memNeeded)
+							if IsResourcesInCluster {
+								isAvailable = true
+								break
+							}
+							time.Sleep(2 * time.Second)
+						}
+						//workaroud for current resources in CI
+						if cpuLimit >= 1 && memLimit >= 1 {
+							Expect(isAvailable).To(BeTrue(), "Cluster should have enough resources")
+						} else {
+							break
+						}
+
+						vm_name := "vm13-" + strconv.Itoa(vm_index)
+						vm_index++
+
+						filename := getYAMLFilename(sockets, cores, threads, address_common)
+						arguments := getArguments(vm_name, sockets, cores, threads)
+
+						virtRawVMFilePath := address_common + "/sockets_cores_threads_raw_manifest_" + vm_name + ".yaml"
+						tframework.ProcessTemplateWithParameters(filename, virtRawVMFilePath, arguments...)
+
+						By("1.3 Starting gouroutine to create, start and test VM")
+						wg.Add(1)
+						//prevent server overloading - CI workaround (i/o error)
+						time.Sleep(15 * time.Second)
+
+						go func(sockets int, cores int, threads int, wg *sync.WaitGroup) {
```

Comment:
Are you really running all the combinations in paralel? And why don't you use the global sockets, cores, threads and wg? Are the arguments to this method really necessary? The code is slighly unreadable because of them and the necessary parentheses block at line 485.

Can some golang expect check this please?

Reply:
>Are you really running all the combinations in paralel?

I don't understand what happen, but if I don't use timeout, the code panics. It runs about 4-5 VMs in one time by fact.

Reply:
I will do additional research: decrease timeout. And temporary delete network-related tests from PR

Reply:
perhaps you want to move this function definition elsewhere and just call it here?
Martin is right, it's harder to read this way.

You can just define this in the top of the test `runMyVM := func(sockets int, cores int, threads int, wg*sync.WaitGroup) {`
Then call it here go runMyVM(sockets, cores, threads, &wg)

Reply:
Vadik, thank you. I've Implemented it.